logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -494,39 +493,42 @@ internal class ConversationDataSource internal constructor(
 
     override suspend fun addMembers(
         userIdList: List<UserId>,
-        conversationID: ConversationId
-    ): Either<CoreFailure, MemberChangeResult> {
-        val users = userIdList.map {
-            idMapper.toApiModel(it)
+        conversationId: ConversationId
+    ): Either<CoreFailure, MemberChangeResult> =
+        detailsById(conversationId).flatMap { conversation ->
+            when (conversation.protocol) {
+                is Conversation.ProtocolInfo.Proteus ->
+                    addMembersToCloudAndStorage(userIdList, conversationId)
+
+                is Conversation.ProtocolInfo.MLS -> {
+                    mlsConversationRepository.addMemberToMLSGroup(conversation.protocol.groupId, userIdList)
+                        .map { MemberChangeResult.Changed(Clock.System.now().toString()) }
+                }
+            }
         }
-        val addParticipantRequest = AddConversationMembersRequest(users, DEFAULT_MEMBER_ROLE)
-        return addedMemberFromCloudAndStorage(userIdList, addParticipantRequest, conversationID)
-    }
 
-    private suspend fun addedMemberFromCloudAndStorage(
-        userIdList: List<UserId>,
-        addParticipantRequest: AddConversationMembersRequest,
-        conversationId: ConversationId
-    ) =
+    private suspend fun addMembersToCloudAndStorage(userIdList: List<UserId>, conversationId: ConversationId) =
         wrapApiRequest {
-            conversationApi.addMember(addParticipantRequest, idMapper.toApiModel(conversationId))
-        }.fold({
-            Either.Left(it)
-        }, { response ->
-            wrapStorageRequest {
-                userIdList.map { userId ->
-                    // TODO: mapping the user id list to members with a made up role is incorrect and a recipe for disaster
-                    Conversation.Member(userId, Conversation.Member.Role.Member)
-                }.let { membersList ->
-                    persistMembers(membersList, conversationId)
-                }
-            }.map {
+            val users = userIdList.map {
+                idMapper.toApiModel(it)
+            }
+            val addParticipantRequest = AddConversationMembersRequest(users, DEFAULT_MEMBER_ROLE)
+            conversationApi.addMember(
+                addParticipantRequest, idMapper.toApiModel(conversationId)
+            )
+        }.flatMap { response ->
+            val memberList = userIdList.map { userId ->
+                // TODO: mapping the user id list to members with a made up role is incorrect and a recipe for disaster
```

Comment:
don't we receive the member role here from the API? shouldn't we address the above todo here?

Reply:
We should, but I'd argue not in this PR.